src/realm/sync/network/default_socket.hpp

```diff
@@ -84,27 +80,88 @@ class EZObserver {
     ~EZObserver() = default;
 };
 
-class EZSocket {
+class DefaultSocketProvider : public SyncSocketProvider {
 public:
-    virtual ~EZSocket();
+    class Timer : public SyncSocketProvider::Timer {
+    public:
+        Timer(network::Service& service, std::chrono::milliseconds delay, FunctionHandler&& handler)
```

Comment:
Should we make this constructor protected so only the socket provider can construct a timer?

Reply:
Done